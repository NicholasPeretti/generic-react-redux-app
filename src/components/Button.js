import React from 'react'
import Style from '../style'
import styled from 'styled-components'
import Text from './Text'

const StyledButton = styled.button`
  background: ${Style.colors.accent};
  width: 128px;
  height: 32px;
  border: 1px solid ${Style.colors.accent};
  border-radius: 3px;
  color: white;
  cursor: pointer;

  &:active {
    background: ${Style.colors.accentPressed};
  }
`

const Button = ({ children }) => (
  <StyledButton>
    <Text>{children}</Text>
  </StyledButton>
)

export default Button
