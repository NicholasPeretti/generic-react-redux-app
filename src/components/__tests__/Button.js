import React from 'react'
import { shallow } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import Button from '../Button'

describe('Button', () => {
  PresentationalTest(Button)

  test('Should match snapshot', () => {
    const wrapper = shallow(<Button/>)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
