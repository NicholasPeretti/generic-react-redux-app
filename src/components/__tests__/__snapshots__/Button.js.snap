// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `"<button class=\\"sc-bwzfXH czzbfx\\"><span class=\\"sc-bdVaJa bkmfbd\\"></span></button>"`;
